### DISCUSSION TITLE
How to use BaseValidator on Custom Dataset for OpenVINO Model

### DISCUSSION THREAD

**Discussion by Zakihran13** [ultralytics/ultralytics#3512] [Q&A]
Hai,
i just wondering if you guys can help me, recently i use OpenVINO to play with YOLOv8n on custom object detection dataset. i want to validate the model in form of IR format with function that proposed in yolov8 tutorial from OpenVINO. 
![image](https://github.com/ultralytics/ultralytics/assets/80832245/c7dd8c8b-65fd-425b-b1ce-75a57f95a80e)

but, turns out my model cannot use the validatorclass module cause of custom dataset. here i try to modify the function using BaseValidator from yolo.engine module. it can pretty well running but returning empty value. can anyone help me whats going on in my code?
![image](https://github.com/ultralytics/ultralytics/assets/80832245/995ac556-ba1c-40fa-8b9c-476195855bba)

and when i try to retest again, it returning this error even tho i already updated the library...
![image](https://github.com/ultralytics/ultralytics/assets/80832245/4aad16d6-eb8e-4847-ab4a-a20ad3cb6757)

thank you for helping me out...

**Comment by glenn-jocher** [comment]
Just validate normally:

yolo val model=path/to/ov/model data=path/to/data.yaml

**Reply by Zakihran13** [reply]
but i want to validate in the openvino format, is it still working?

**Reply by Zakihran13** [reply]
![image](https://github.com/ultralytics/ultralytics/assets/80832245/223432f5-f902-4d6c-ab59-79e17e882862)

i guess it works. thank you so much for helping me 😁😁

**Reply by glenn-jocher** [reply]
@Zakihran13 you're welcome! I'm glad to hear that it worked out for you. If you have any more questions or need further assistance, feel free to reach out. Happy detecting! 😊🚀
